controllers/restore_resource_version_priority.go

```diff
@@ -0,0 +1,52 @@
+package controllers
+
+import (
+	"fmt"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	enableApiGroupVersionsFeatureFlag      = "EnableAPIGroupVersions"
+	enableApiGroupVersionsConfigMapName    = "enableapigroupversions"
+	restoreResourcesVersionPriorityDataKey = "restoreResourcesVersionPriority"
+)
+// If RestoreResourcesVersionPriority is defined, configmap is created or updated and feature flag for EnableAPIGroupVersions is added to velero
+func (r *VeleroReconciler) ReconcileRestoreResourcesVersionPriority(velero *oadpv1alpha1.Velero) (bool, error) {
+	if len(velero.Spec.RestoreResourcesVersionPriority) == 0 {
+		return true, nil
+	}
+	// if the RestoreResourcesVersionPriority is specified then ensure feature flag is enabled for enableApiGroupVersions
+	// duplicate feature flag checks are done in ReconcileVeleroDeployment
+	velero.Spec.VeleroFeatureFlags = append(velero.Spec.VeleroFeatureFlags, enableApiGroupVersionsFeatureFlag)
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      enableApiGroupVersionsConfigMapName,
+			Namespace: velero.Namespace,
+		},
+	}
+	// Create ConfigMap
+	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, &configMap, func() error {
+		if err := controllerutil.SetControllerReference(velero, &configMap, r.Scheme); err != nil {
+			return err
+		}
+		configMap.Data = make(map[string]string, 1)
```

Comment:
Won't this always overwrite the configmap.Data field? Would it better to check if it's `nil` first?

Reply:
How would you suggest I handle the case if it's not nill? Don't we need to follow whatever CR says?

Reply:
Analyze current configmap Data and append to it maybe.
If there is a conflicting key which already exists, the corresponding values in the CR take precedence

Reply:
Alright, good points. fine by me